### PR TITLE
PAYARA-3723 Fixing sonar blocker bugs with unclosed resources and potential NPE

### PR DESCRIPTION
**Pull request by svendiedrichsen**

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/jws/boot/JWSACCMain.java [hunk 1]
```diff
@@ -40,27 +40,22 @@
...
-import java.io.File;
-import java.io.FileNotFoundException;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.OutputStreamWriter;
+import org.glassfish.appclient.client.acc.AppClientContainer;
+import org.glassfish.appclient.client.acc.JWSACCClassLoader;
+import org.glassfish.appclient.common.Util;
+
+import javax.swing.*;
```
Comment (Pandrex247) [inline appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/jws/boot/JWSACCMain.java]: Minor point, but is this necessary? Aren't we only using one class from here?

File: appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/util/NamingUtilsImpl.java [hunk 2]
```diff
@@ -107,42 +106,41 @@ public NamingObjectFactory createDelegatingNamingObjectFactory(String name,
...
+    private Object deserialize(byte[] data) throws IOException, java.security.PrivilegedActionException {
+        try(final ObjectInputStream ois =
+                getFactory().createObjectInputStream(new ByteArrayInputStream(data))){
+            return AccessController.doPrivileged((PrivilegedExceptionAction) () -> ois.readObject());
+        }
+    }
+
+    private byte[] serialize(Object obj) throws IOException {
+        try(ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ObjectOutputStream oos = getFactory().createObjectOutputStream(bos)) {
```
Comment (Pandrex247) [inline appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/util/NamingUtilsImpl.java]: Formatting is a little off here, makes this a bit hard to read as this line looks like it belongs with the ones below.

File: appserver/ejb/ejb-http-remoting/endpoint/src/main/java/fish/payara/ejb/invoke/InvokeEJBServlet.java [hunk 3]
```diff
@@ -167,47 +170,49 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
             throw new IllegalStateException(e);
         }
     }
-    
+
     private Object toObject(JsonValue objectValue, Class<?> type) {
-        return JsonbBuilder
-                .create()
-                .fromJson(objectValue.toString(), type);
+        try(Jsonb jsonb = JsonbBuilder.create()) {
```
Comment (Pandrex247) [inline appserver/ejb/ejb-http-remoting/endpoint/src/main/java/fish/payara/ejb/invoke/InvokeEJBServlet.java]: Super minor, you missed a space between `try` and `(`

File: appserver/payara-appserver-modules/security-openid/src/main/java/fish/payara/security/openid/OpenIdAuthenticationMechanism.java [hunk 4]
```diff
@@ -245,6 +243,12 @@ private AuthenticationStatus validateAuthorizationCode(HttpMessageContext httpCo
         }
     }
 
+    private JsonObject readJsonObject(String tokensBody) {
+        try(JsonReader reader = Json.createReader(new StringReader(tokensBody))) {
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/security-openid/src/main/java/fish/payara/security/openid/OpenIdAuthenticationMechanism.java]: Same as other comment, just nitpicking formatting here

**Comment by svendiedrichsen** [comment]
@Pandrex247 I have made the requested code changes.

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/jws/boot/JWSACCMain.java (hunk 1 above)
Comment (svendiedrichsen) [reply appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/jws/boot/JWSACCMain.java]: done

File: appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/util/NamingUtilsImpl.java (hunk 2 above)
Comment (svendiedrichsen) [reply appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/util/NamingUtilsImpl.java]: This is actually 2 Closables in one try-with-resources block. To have them both automatically closed. I have made now 2 try blocks to make clearer.

**Comment by Cousjava** [comment]
Jenkins test please

**Pull request merged**